src/main/java/io/vertx/mqtt/MqttEndpoint.java

```diff
@@ -393,6 +552,12 @@
    */
   Future<Integer> publish(String topic, Buffer payload, MqttQoS qosLevel, boolean isDup, boolean isRetain, int messageId);
 
+  /**
+   * Like {@link #publish(String, Buffer, MqttQoS, boolean, boolean, int, properties, Handler)} but returns a {@code Future} of the asynchronous result
+   */
+  @GenIgnore
+  Future<Integer> publish(String topic, Buffer payload, MqttQoS qosLevel, boolean isDup, boolean isRetain, int messageId, MqttProperties properties);
```

Comment:
we need the matching method with `Handler<AsyncResult<Integer>>` and use `PERMITTED_TYPE` in `@GenIgnore`